### PR TITLE
feat(WEB-821): Add onScreenChanged callback to widget and popup

### PR DESCRIPTION
**Pull request by AlfonsoCerrato**
In order to track analytics events once the user interacts with the embedded Typeform, I introduced a callback `onScreenChanged` activated by the `form-screen-changed` event introduced with this [PR](https://github.com/Typeform/stakhanov/pull/3936) on stakhanov.

### CODE REVIEW

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/45e901b6d9f495aad1ea1fd7515b9d0fc2dec0a9/embed.js

File: src/core/views/widget.js [hunk 1]
```diff
@@ -191,6 +194,12 @@ class Widget extends Component {
     })
   }
 
+  handleFormScreenChanged (event) {
```
Comment (mathio) [inline src/core/views/widget.js:197]: What about popup embed type? I think it would be beneficial to have this event for both widget and embed.

**Review by mathio — COMMENTED** [review]
Can you please write unit tests for this? See tests for `onSubmit` and `onAutoClose` in `src/core/views/widget.spec.js` and `src/core/views/popup.spec.js`.

File: src/core/views/widget.js (hunk 1 above)
Comment (AlfonsoCerrato) [reply src/core/views/widget.js:197]: Hi @mathio, thanks for the feedback. I just added the same functionality to the popup as well.

**Review by mathio — APPROVED** [review]
Looks good! ✅  We can release this tomorrow morning if you want.

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/ac1484633f2211f27c3bdd966599c4b4efc7ccbc/embed.js

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/33955332f3718308dbf0f05a69a0ecc4e81aaa51/embed.js

**Pull request merged**